module/apmgopgv10/hook.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.14
+
+package apmgopgv10 // import "go.elastic.co/apm/module/apmgopgv10"
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-pg/pg/v10"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmsql"
+	"go.elastic.co/apm/stacktrace"
+)
+
+func init() {
+	stacktrace.RegisterLibraryPackage("github.com/go-pg/pg/v10")
+}
+
+const elasticApmSpanKey = "go-apm-agent:span"
+
+// Instrument modifies db such that operations are hooked and reported as spans
+// to Elastic APM if they occur within the context of a captured transaction.
+//
+// If Instrument cannot instrument db, then an error will be returned.
+func Instrument(db *pg.DB) error {
+	qh := &queryHook{}
+	switch qh := ((interface{})(qh)).(type) {
+	case pg.QueryHook:
+		db.AddQueryHook(qh)
+		return nil
+	}
+	return errors.New("cannot instrument pg.DB, does not implement required interface")
+}
+
+// queryHook is an implementation of pg.QueryHook that reports queries as spans to Elastic APM.
+type queryHook struct{}
+
+// BeforeQuery initiates the span for the database query
+func (qh *queryHook) BeforeQuery(evt *pg.QueryEvent) {
```

Comment:
```suggestion
func (qh *queryHook) BeforeQuery(ctx context.Context, evt *pg.QueryEvent) (context.Context, error) {
```

Turns out that the [QueryHook](https://pkg.go.dev/github.com/go-pg/pg/v10#QueryHook) interface has changed in v10. Instead of using `evt.Stash` we should return a context with the span, and extract the span from the context passed into `AfterQuery`.